### ISSUE TITLE
UXD: basic user flow for Blank Map

### ISSUE THREAD

**Issue by ryanfchase** [hackforla/311-data#1848]
### Overview 
We need to explore new user flows for a blank map design, so that we can learn what components are needed for when no data is loaded 

### More info
We need to design a flow for page-load where no data is visualized, instead we will prompt the user on how to use the application, which will significantly cut back on the amount of time it takes to open our site and help user retention.

### Action Items

- [x] Create user flow that requires user to specify all filter criteria and confirm it
- [x] Create low-fidelity mockups to specify which components need to be created for the Blank Map user flow
- [x] Verify with dev lead that this user flow is technically feasible 
- [x] Document user interaction in Figma

---

### Hand Off Materials

Figma Section Name: `REPLACE WITH SECTION NAME`

<details><summary>Before Screenshot</summary>
<p>

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 

<details><summary>After Screenshot (Finalized)</summary>
<p>

Basic User Flow

![Image](https://github.com/user-attachments/assets/5839280c-9b81-4b82-bf04-3f53459e3713)

Combined Acknowledgement Modal
<img width="692" alt="Image" src="https://github.com/user-attachments/assets/062d8b1e-9925-4722-ab8e-a06ff1224103" />

Empty Filter Criteria in Search/Filters Modal
<img width="668" alt="Image" src="https://github.com/user-attachments/assets/f1bf3895-106c-4ec9-bcd1-31f42825db3d" />

Loading Bar Modal
<img width="693" alt="Image" src="https://github.com/user-attachments/assets/9a10ad1a-550f-4b55-ae1b-789ab28d1e0d" />

</p>
</details> 

### Designer Resources

<details><summary>Iteration Dropdown Copy/Paste</summary>
<p>

```
<details><summary>Iteration X</summary>
<p>

Link to notes: `REPLACE WITH COMMENT URL`

`REPLACE WITH SCREENSHOT UPLOAD`

</p>
</details> 
```

</p>
</details> 

<details><summary>Instructions for Engineering Hand Off</summary>
<p>

To Start Engineering Hand Off...
1. Ensure all Hand Off Materials are filled in
3. Add the "ready for dev lead" label
4. Leave a comment saying "This ticket is ready for engineering hand off." 

</p>
</details>

**Comment by ryanfchase** [comment]
Coming from Bonnie, this is a `Pre-Launch` ticket. Please try to prioritize discussing and advancing this ticket as soon as you are able. Thanks.

**Comment by Joy-Truex** [comment]
This feels like a larger project for pre-launch in my opinion, I think this needs further discuss (although I see the discussion label on it).

I feel to do an adequate job this feels like an additional feature rather than simply a modal-- feels like it is asking for an onboarding/tutorial flow that may not be able to be addressed by one screen and would require user testing to ensure that it is adequately sufficient. Can we discuss more at our next meeting?

**Comment by ryanfchase** [comment]
> This feels like a larger project for pre-launch in my opinion, I think this needs further discuss (although I see the discussion label on it).
> 
> I feel to do an adequate job this feels like an additional feature rather than simply a modal-- feels like it is asking for an onboarding/tutorial flow that may not be able to be addressed by one screen and would require user testing to ensure that it is adequately sufficient. Can we discuss more at our next meeting?

Sure! This has been added to Topic Requests for 2024-10-23 Weekly meeting.

**Comment by Joy-Truex** [comment]
@ryanfchase Upon discussing and reviewing with the team at our last meeting, our feedback is that this was previously addressed by [Issue 1655](https://github.com/hackforla/311-data/issues/1655) with the loading modal- any other small solution would likely have a similar impact to that 
Can you review with PM team and let us know if you have further thoughts on how to proceed or need more clarification? TY!

**Comment by ryanfchase** [comment]
I will be assigning myself while I provide some data visualizations mockups that only work with a Blank Map on page load. No action needed from design for now.

**Comment by ryanfchase** [comment]
Adding this for agenda for tonight. I want to see if dev lead has something in mind, since we are working on updates to our Performance Epic: https://github.com/hackforla/311-data/issues/1890

**Comment by ryanfchase** [comment]
This discussion has been kicked back to the Product Planning meeting

**Comment by ryanfchase** [comment]
Moved out of icebox (it was awaiting discussion in Product Planning). Question for design lead: does this need any further information or updates? Let me know if I need to add anything that was discussed in Product Planning in order for you to proceed.

**Comment by Joy-Truex** [comment]
@ryanfchase I believe this is part of this ticket and not separate-- I would like to add the need for a final user flow for this

**Comment by ryanfchase** [comment]
> I believe this is part of this ticket and not separate-- I would like to add the need for a[ final user flow](https://www.figma.com/design/eF63O8yIGz2soQ4JJFfyoK/311_Data_UXUI?node-id=16568-860&t=IuHG0SSm7CsIDy4i-1) for this

Thank you @Joy-Truex! I'm providing a screengrab so people can see this basic user flow at a glance (although, glancing is not a substitute for a full read-through of each design and dev note. Other readers be warned ⚠).

<details><summary>Screengrab of Blank Map user flow</summary>
<p>

![Image](https://github.com/user-attachments/assets/5839280c-9b81-4b82-bf04-3f53459e3713)

</p>
</details> 

I don't see any further questions for product. Moving to New Issue Approval with `draft` assigned to you while you prepare this ticket for design. If this ticket requires more extensive work than our normal design tickets allot, we can work together to modify this ticket accordingly.

**Comment by Joy-Truex** [comment]
Will be discussed at breakout room of 2-12-25 general meeting. To update ticket with notes after meeting.

**Comment by ryanfchase** [comment]
FYI Moving to questions, since it won't receive updates until after being discussed. @Joy-Truex thank you for using `on agenda: general`, that was the correct label to use. I can get this clarified, but New Issue Approval should mean "ticket is waiting on input from lead". PMs can move this back to New Issue Approval after tonight's meeting.

**Comment by ryanfchase** [comment]
This agenda item was discussed at the weekly meeting. Moving back to New Issue Approval.

**Comment by Joy-Truex** [comment]
have solicited feedback from design team. will provide an update on steps forward prior to next GM. ETA 2/19

**Comment by Joy-Truex** [comment]
Lofi to be discussed at GM breakout

<details><summary>1. Welcome to 311 Data message</summary>
<p>

- Consider adapting the loading modal's language to signify that we are just preparing the map (and not loading any data)
- Consider adapting the acknowledgement modal to provide some basic instructions on how to use the Search & Filters Modal
- instead of doing two separate pop-ups, combine them into one that is two pages;
with next and then ok

<img width="692" alt="Image" src="https://github.com/user-attachments/assets/062d8b1e-9925-4722-ab8e-a06ff1224103" />

</p>
</details> 

<details><summary>2. blank map & blank filters 3. confirm</summary>
<p>

- no address filled in
- no boundaries (NC)
- no date range
- no request type
- no request status

- confirm so that the map doesn’t load “by accident” thus wasting time

<img width="668" alt="Image" src="https://github.com/user-attachments/assets/f1bf3895-106c-4ec9-bcd1-31f42825db3d" />

</p>
</details> 

<details><summary>4. loading modal</summary>
<p>

- explains what is happening with more of a countdown or more indication of what is happening
- for example when you’re downloading a large file

<img width="693" alt="Image" src="https://github.com/user-attachments/assets/9a10ad1a-550f-4b55-ae1b-789ab28d1e0d" />

</p>
</details>

**Comment by Joy-Truex** [comment]
This ticket can be marked as completed/closed. Thanks!
